packages/README.md

```diff
@@ -0,0 +1,73 @@
+# Package Metadata Description
+
+## 1. Abstract
+Working with many differing artifacts coming out of many differing package ecosystems requires us to find common ground between all
+the ecosystems. While *requires* may have been a little strong wording to start with, let's overview the benefits of having
+a common metadata description for all ecosystems:
+
+ * Metadata storage - Database
+   * All data are stored in generic rows
+   * There' onlys one main table
+   * Database can operate with the metadata directly
+ * Common codebase
+   * One API, one Interchange format
+   * One binding for all client languages
+ * Upstreams can change their metadata format on a whim
+
+While some points are important during development, especially the last bullet is very important for long term maintenance, since it 
+allows us to `contain` the volatile upstream specification within single minimal component for each ecosystem. Let's see an example:
+
+```
+         +-------------------+
+IN       |                   |  OUT
+         |  npm metadata     |
++------->+  manifest tool    +------------>
+         |                   |
+         +-------------------+
+```
+
+`IN`put is a `npm` package and `OUT`put is a document in common metadata format. Whenever the `npm` upstream changes its package specification
+we only need to update this single component. The rest of the infrastructure is completely oblivious to the changes because it already
+consumes the common metadata format.
+
+Furthermore, taking a service oriented architecture into the picture, the crucial aspect of properly decoupled
+services is *the means* how the services talk to each other, how they communicate, be it either the APIs or the underlying data interchange format.
+
+## 2. Expressivity and flexibility
+
+The language of the common metadata format shall be expressive enough so that we can always express a feature of one package format in terms
+of the common metadata format. Expressivity can be thought of as the degree of *freedom* or *possibilities* that is needed to describe the particular
+metadata component. Let's illustrate this on versioned dependencies - while some package ecosystem can do fine with rather simplistic rules for dependency versioning:
+
+  * `>= 1.0.0`
+  * `1.0.0`
+  * `>= 1.2`
+  
+There also exist much *more complex* dependency variants:
```

Comment:
How far do we want to go here? https://docs.npmjs.com/files/package.json#dependencies